coderd/database/dbmem/dbmem.go

```diff
@@ -155,6 +158,7 @@ type data struct {
 	groups                        []database.Group
 	jfrogXRayScans                []database.JfrogXrayScan
 	licenses                      []database.License
+	notificationMessages          []database.NotificationMessage
```

Comment:
Self-review: this should be `[]*database.NotificationMessage` since `AcquireNotificationMessages` is mutating the entries.